engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportToSizeOf.java

```diff
@@ -65,12 +69,20 @@ public RenderPipelineTask generateTask() {
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(fboName);
+        // Generates a unique 32 bit signed integer from two integers.
+        // This will return unique values for the following reasonable ranges:
+        // width < (1 << 16) (65536) and height < (1 << 15) (32768)
+        return getWidth() << 15 | getHeight();
     }
 
     @Override
     public boolean equals(Object obj) {
-        return (obj instanceof SetViewportToSizeOf) && this.fboName.equals(((SetViewportToSizeOf) obj).getFboName());
+        if (!(obj instanceof SetViewportToSizeOf))
+            return false;
+
+        SetViewportToSizeOf other = (SetViewportToSizeOf) obj;
+
+        return getWidth() == other.getWidth() && getHeight() == other.getHeight();
```

Comment:
I think this is cleaner and _much_ more easily understandable than casting `obj` to `SetViewportToSizeOf` twice in a huge boolean statement.